### PR TITLE
[ElasticDL] Generate data conversion script

### PR DESCRIPTION
**Pull request by terrytangyuan**
Part of #664, including:
* Add `getFeaturesNames` to obtain the feature names.
* Add `makePythonListCode ` to create Python list code from Go list of strings.
* Generate data conversion script in `elasticDLTrain `.

### CODE REVIEW

**Comment by terrytangyuan** [comment]
The Travis build failure is not related to this PR. I’ve verified that the relevant tests pass locally.

File: sql/codegen_elasticdl.go [hunk 1]
```diff
@@ -48,14 +49,48 @@ type elasticDLFiller struct {
 	TrainClause *resolvedTrainClause
 }
 
-func newElasticDLDataConversionFiller(odpsTableName string, featuresList string, batchSize int, numProcesses int) (*elasticDLDataConversionFiller, error) {
+func getFeaturesNames(pr *extendedSelect) ([]string, error) {
```
Comment (typhoonzero) [inline sql/codegen_elasticdl.go:52]: Please be aware that not all selected columns are used as features for training, we can write `select * from some_table train MyModel column a label x ...` and the table may have columns `a,b,c,d...` but only a will be used as training input.

Get feature list from `COLUMN` clause or you can just write some comment here for later update.

File: sql/codegen_elasticdl.go [hunk 2]
```diff
@@ -82,32 +117,51 @@ func newElasticDLTrainFiller(pr *extendedSelect, db *DB, session *pb.Session, ds
...
+	// TODO: Execute the script inside container where ElasticDL is available
+	var dataConversionProgram bytes.Buffer
+	dataConversionFiller, err := newElasticDLDataConversionFiller(pr, 200, 1)
+	if err != nil {
+		return err
+	}
+	if err = elasticdlDataConversionTemplate.Execute(&dataConversionProgram, dataConversionFiller); err != nil {
+		return fmt.Errorf("Failed executing data conversion template: %v", err)
+	}
+	dataConversionScriptPath := "data_conversion.py"
```
Comment (typhoonzero) [inline sql/codegen_elasticdl.go:130]: File `data_conversion.py` is not included in this PR?

**Review by typhoonzero — APPROVED** [review]

File: sql/codegen_elasticdl.go (hunk 1 above)
Comment (terrytangyuan) [reply sql/codegen_elasticdl.go:52]: Yes, this issue will be tracked in the separate TODO issue.

File: sql/codegen_elasticdl.go (hunk 2 above)
Comment (typhoonzero) [reply sql/codegen_elasticdl.go:130]: Sorry did not see how it's generated.

**Review by tonyyang-svail — APPROVED** [review]
The CI failure is not relevant to this PR. I am approving and merging this PR.

I will find out the cause of the CI failure in a separate PR.

**Pull request merged**